test/e2e/azure_privatecluster.go

```diff
@@ -458,3 +456,22 @@ func getAPIVersionFromUserAgent(userAgent string) string {
 	splits := strings.Split(userAgent, "/")
 	return splits[len(splits)-1]
 }
+
+func getClientIDforMSI(resourceID string) string {
```

Comment:
Is it reasonable to re-use the MSI `GetClientID` func in azure/services/identities/client.go?

Reply:
I tried, that was my first attempt actually. But it turned out to be pretty difficult without doing a bunch of code refactoring because the client is private and the only way to create a new one is to have a infrav1 Authorizer interface which is different from the authorest Authorizer :( 

In the end I figured the complexity was low enough that it wasn't too bad to duplicate and since this is part of the test it might be better not to rely on the code we are testing as part of the test setup anyways.